docs/markdown/Features-module.md

```diff
@@ -0,0 +1,320 @@
+)
+## ARMv8.2 Single & half-precision Multiply
+ASIMDFHM = features.new(
+  'ASIMDFHM', 4, implies: ASIMDHP,
+  args: {'val': '-march=armv8.2-a+fp16fml', 'match': '-march=.*', 'mfilter': '\+.*'}
+)
+```
+### features.test()
+**Signature**
+```meson
+[bool, Dict] features.test(
+  # Positional arguments:
+  FeatureObject...
+  # Keyword arguments:
+  anyfet      : bool = false,
+  force_args  : str | list[str] | empty = empty,
+  compiler    : Compiler | empty = empty,
+  cached      : bool = true
+)
+```
+
+Test one or a list of features against the compiler and retrieve a dictionary containing
+essential information required for compiling a source that depends on these features.
+
+A feature is deemed supported if it fulfills the following criteria:
+
+ - It is not marked as disabled.
+ - The compiler accommodates the features arguments.
+ - Successful compilation of the designated test file or code.
+ - The implied features are supported by the compiler, aligning with
+   the criteria mentioned above.
+
+This function requires at least one feature object as positional argument,
+and additional keyword arguments are elaborated as follows:
+
+- `anyfet` **bool = false**:
+  If set to true, the returned dictionary will encompass information regarding
+  the maximum features available that are supported by the compiler.
+  This extends beyond the specified features and includes their implied features.
+- `force_args` **str | list[str] | empty = empty**:
+  An optional single compiler argument or an array of compiler arguments to be
+  employed instead of the designated features' arguments when testing the
+  test code against the compiler.  This can be useful for detecting host features.
+- `compiler` **Compiler | empty = empty**:
+  A Compiler object to be tested against. If not defined,
+  the function will default to the standard C or C++ compiler.
+- `cached`: **bool = true**:
+  Enable or disable the cache. By default, the cache is enabled,
+  enhancing efficiency by storing previous results.
+
+This function returns a list of two items. The first item is a boolean value,
+set to true if the compiler supports the feature and false if it does not.
+The second item is a dictionary that encapsulates the test results, outlined as follows:
+
+**structure**
+```meson
+{
+    'target_name'        : str,
+    'prevalent_features' : list[str],
+    'features'           : list[str],
+    'args'               : list[str],
+    'detect'             : list[str],
+    'defines'            : list[str],
+    'undefines'          : list[str],
+    'is_supported'       : bool,
+    'is_disabled'        : bool,
+    'fail_reason'        : str
+}
+```
+
+### features.multi_targets()
+**Signature**
+```meson
+TargetsObject features.multi_targets(
+  # Positional arguments:
+  str,
+  (
+    str | File | CustomTarget | CustomTargetIndex |
+    GeneratedList | StructuredSources | ExtractedObjects |
```

Comment:
Are StructuredSources necessary here?

They are only permitted in the first place, in:
- Java jars
- rust targets (cannot be polyglot languages in the same target, you have to have a pure rust target that can be linked with other languages as a static/shared library)

So supporting it here feels premature.